metrics/prometheus_machine.go

```diff
@@ -192,9 +192,58 @@ func NewPrometheusMachineCollector(i infoProvider, includedMetrics container.Met
 			},
 		}...)
 	}
+
+	if includedMetrics.Has(container.PerfMetrics) {
+		c.machineMetrics = append(c.machineMetrics, []machineMetric{
+			{
+				name:        "machine_perf_uncore_metric",
+				help:        "Perf uncore event metric.",
+				valueType:   prometheus.CounterValue,
+				extraLabels: []string{"node", "event", "type"},
+				getValues: func(machineInfo *info.MachineInfo) metricValues {
+					values := make(metricValues, 0, len(machineInfo.PerfUncoreStats))
+					for _, metric := range machineInfo.PerfUncoreStats {
+						values = append(values, metricValue{
+							value:     float64(metric.Value),
+							labels:    []string{getNUMANode(metric.Cpu, machineInfo.Topology), metric.Name, strconv.FormatUint(uint64(metric.Type), 10)},
+							timestamp: machineInfo.Timestamp,
+						})
+					}
+					return values
+				},
+			},
+			{
+				name:        "machine_perf_uncore_metric_scaling_ratio",
+				help:        "Perf uncore event metric scaling ratio.",
+				valueType:   prometheus.GaugeValue,
+				extraLabels: []string{"node", "event", "type"},
+				getValues: func(machineInfo *info.MachineInfo) metricValues {
+					values := make(metricValues, 0, len(machineInfo.PerfUncoreStats))
+					for _, metric := range machineInfo.PerfUncoreStats {
+						values = append(values, metricValue{
+							value:     metric.ScalingRatio,
+							labels:    []string{getNUMANode(metric.Cpu, machineInfo.Topology), metric.Name, strconv.FormatUint(uint64(metric.Type), 10)},
+							timestamp: machineInfo.Timestamp,
+						})
+					}
+					return values
+				},
+			},
+		}...)
+	}
+
 	return c
 }
 
+func getNUMANode(cpu int, nodes []info.Node) string {
```

Comment:
To have an order in this file I suggest moving this function below, under all functions with PrometheusMachineCollector receiver, near getMemoryByType function.

Reply:
Good idea, I also provided unit tests.